src/main/java/com/direwolf20/buildinggadgets/common/save/SaveManager.java

```diff
@@ -0,0 +1,84 @@
+package com.direwolf20.buildinggadgets.common.save;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.config.Config;
+import com.direwolf20.buildinggadgets.common.util.ref.Reference.SaveReference;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.storage.WorldSavedData;
+import net.minecraftforge.fml.event.server.FMLServerStartedEvent;
+import net.minecraftforge.fml.event.server.FMLServerStoppedEvent;
+
+import java.util.function.IntSupplier;
+import java.util.function.Supplier;
+
+public enum SaveManager {
+    INSTANCE;
+    private SaveTemplateProvider templateProvider;
+    private TemplateSave templateSave;
+    private UndoWorldSave copyPasteUndo;
+    private UndoWorldSave destructionUndo;
+    private UndoWorldSave buildingUndo;
+    private UndoWorldSave exchangingUndo;
+
+    SaveManager() {
+        this.templateProvider = new SaveTemplateProvider(this::getTemplateSave);
+    }
+
+    public void onServerStarted(FMLServerStartedEvent event) {
+        BuildingGadgets.LOG.debug("Loading World Saves.");
+        ServerWorld world = event.getServer().getWorld(DimensionType.OVERWORLD);
+        copyPasteUndo = getUndoSave(world, Config.GADGETS.GADGET_COPY_PASTE.undoSize::get, SaveReference.UNDO_COPY_PASTE);
```

Comment:
What if the config changes whilst the game is open using the config gui?

Reply:
The config Gui isn't up yet. You'd have to do that by changing the File...

As I said on Discord, as soon as you access an UndoHistory, it will try to evict everything that goes beyond the current config limit. Therefore they'd be deleted at the moment.

Reply:
Awesome! Just wanted to make sure.